qml/pages/SatelliteInfoPage.qml

```diff
@@ -114,11 +116,62 @@ Page {
+                        opacity: (declination !=0) ? 1 : 0 //0.5
+                    }
+        //Movement Direction line
+            Rectangle { visible:  !isNaN(gpsDataSource.movementDirection)
+                        width: Theme.iconSizeExtraSmall / 10.0
+                        height: diameter/2
+                        anchors.left: parent.horizontalCenter
+                        anchors.bottom: parent.verticalCenter  //+diameter/2
+                        transform: Rotation { origin.x: 0 ; origin.y: diameter/2; angle: isNaN(gpsDataSource.movementDirection) ? 170 : gpsDataSource.movementDirection}
+                        color: "cyan"
+                        opacity: 1//(declination !=0) ? 1 : 0
+                    }
+
+        // North, East, South, West, MagneticNorth indicators
+        Repeater {
+            model: declination !=0 ? ["N", "E", "S", "W", "M"]:["N", "E", "S", "W"]
+
+            delegate:
+                Label {
+                x: center + Math.sin((index !==4 ? (index * 90) : declination) * Math.PI / 180) * (radius+width/2) - width / 2.0
+                y: center - Math.cos((index !==4 ? (index * 90) : declination) * Math.PI / 180) * (radius+width/2) - height / 2.0
+                color: "white"
+                font.weight: Font.Bold
+                font.pixelSize: Theme.fontSizeExtraSmall
+                property int iN: index !==4 ? 0 : compass.reading.calibrationLevel > 0.99 ?1:2
+                text: " "+[modelData,"M","m"][iN]+" "  //index !==4 ? " "+modelData+" " : compass.reading.calibrationLevel > 0.99 ? " M ":" m "
+
+
+                // Negate the radar containers rotation, so that the boxes and texts
+                // stay upright according to device orientation. Note that radar.rotation
+                // value read is already smoothed, so we can just use it raw.
+                rotation: -radar.rotation
 
-        // Satellite identifiers (numbers)
-        // and their respective box color
+                Rectangle {
+                    z: -1
+                    anchors.horizontalCenter: parent.horizontalCenter
+                    anchors.verticalCenter: parent.verticalCenter
+                    width: parent.width + parent.font.pixelSize / 8.0
+                    height: parent.height + parent.font.pixelSize / 8.0
+                    color: ["blue","red","transparent"][iN] //index !== 4 ? "blue" : "red"
+                    radius: parent.font.pixelSize / 8.0
+                }
+            }
+        }
+        // Satellite identifiers (numbers), and their respective box rssi color and inUse border
+        // first draw them all solid , hiding the radar chart background, and ensuring colors are correct
         Repeater {
-            model: status === PageStatus.Inactive ? [] : gpsDataSource.satellites
+            model: status === PageStatus.Inactive ? [] : gpsDataSource.satellites.sort(function(a,b) {return (a.inUse ? 1:-1) - (b.inUse ? 1:-1)})
```

Comment:
Perhaps this sorting could be done in gpsDataSource instead?

Reply:
Can't do that at the QML level, but I have removed the duplicate.
Not sure if it makes sense in the c++, since it's specific to the satinfo page